### DISCUSSION TITLE
【Bug?】The behavior of the UPDATE statement using the CASE clause is wrong

### DISCUSSION THREAD

**Discussion by hiroyuki0415** [MarkMpn/Sql4Cds#313] [Q&A]
I have the following data:
<HTML><head></head>
<BODY>
<TABLE><THEAD><TH>&nbsp;</TH><TH>crf3c_rank0_3id</TH><TH>crf3c_integer_column</TH></THEAD><TR><TD ALIGN="center">&nbsp;</TD><TD>2f0384fb-5b14-ee11-9cbe-002248627e12</TD><TD>1</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>63aecf0d-5c14-ee11-9cbe-002248627e12</TD><TD>1</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>13a93ef8-5b14-ee11-9cbe-002248627ef1</TD><TD>1</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>b51e0406-5c14-ee11-9cbe-002248627ef1</TD><TD>1</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>08a15f0d-5c14-ee11-9cbe-002248627ef1</TD><TD>1</TD></TR></TABLE>

</BODY>
</HTML>

After executing the following UPDATE, all the values ​​in the [crf3c_integer_column] column are NULL.
Isn't this a bug?

```sql
UPDATE r
SET r.crf3c_integer_column = (
		CASE 
			WHEN r.crf3c_integer_column IS NULL
				THEN NULL
			ELSE r.crf3c_integer_column + 1
			END
		)
FROM crf3c_rank0_3 AS r
WHERE r.crf3c_rank0_3id IS NOT NULL;
```

The result is below:

<HTML><head></head>
<BODY>
<TABLE><THEAD><TH>&nbsp;</TH><TH>crf3c_rank0_3id</TH><TH>crf3c_integer_column</TH></THEAD><TR><TD ALIGN="center">&nbsp;</TD><TD>2f0384fb-5b14-ee11-9cbe-002248627e12</TD><TD>NULL</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>63aecf0d-5c14-ee11-9cbe-002248627e12</TD><TD>NULL</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>13a93ef8-5b14-ee11-9cbe-002248627ef1</TD><TD>NULL</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>b51e0406-5c14-ee11-9cbe-002248627ef1</TD><TD>NULL</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>08a15f0d-5c14-ee11-9cbe-002248627ef1</TD><TD>NULL</TD></TR></TABLE>

</BODY>
</HTML>

Thanks in advance for your support.

**Comment by MarkMpn** [comment]
Thanks for the details. Yes, I can confirm this is a bug, I'll try to get this fixed ASAP. In the meantime you can get the correct results either by reversing the order of the `CASE` statement:

```sql
UPDATE r
SET r.crf3c_integer_column = (
		CASE 
			WHEN r.crf3c_integer_column IS NOT NULL
				THEN r.crf3c_integer_column + 1
			ELSE NULL
			END
		)
FROM crf3c_rank0_3 AS r
WHERE r.crf3c_rank0_3id IS NOT NULL;
```

or by removing it entirely (`NULL + 1` is still `NULL`):

```sql
UPDATE r
SET r.crf3c_integer_column = r.crf3c_integer_column + 1
FROM crf3c_rank0_3 AS r
WHERE r.crf3c_rank0_3id IS NOT NULL;
```

I've moved this to issue #314 to track it until it's resolved.

**Comment by hiroyuki0415** [comment]
Thank you for your prompt reply.
